disk-deactivate/disk-deactivate.jq

```diff
@@ -0,0 +1,77 @@
+# since lsblk lacks zfs support, we have to do it this way
+def remove:
+  if .fstype == "zfs_member" then
+    "zpool destroy -f \(.label)"
+  elif .fstype == "LVM2_member" then
+    [
+      "vg=$(pvs \(.path) --noheadings --options vg_name | grep -o '[a-zA-Z0-9-]*')",
+      "vgchange -a n \"$vg\"",
+      "vgremove -f \"$vg\""
+    ]
+  elif .fstype == "swap" then
+    "swapoff \(.path)"
+  elif .fstype == null then
+    # maybe its zfs
+    [
+      # the next line has some horrible escaping
+      "zpool=$(zdb -l \(.path) | sed -nr $'s/ +name: \\'(.*)\\'/\\\\1/p')",
```

Comment:
What argument are you passing to `zdb` here? Is this the device name?

Reply:
the device path, for example '/dev/sda3' or '/dev/mapper/crypted'. we do this to find out if the device is part of a zpool and get the zpool name then. a typical zdb -l output looks like this:
```
------------------------------------
LABEL 0
------------------------------------
    version: 5000
    name: 'rpool'
    state: 0
    txg: 26685541
    pool_guid: 6968731913350947701
    errata: 0
    hostid: 579054254
    hostname: 'prism'
    top_guid: 185435801962358030
    guid: 8148307730435269824
    vdev_children: 1
    vdev_tree:
        type: 'mirror'
        id: 0
        guid: 185435801962358030
        metaslab_array: 256
        metaslab_shift: 31
        ashift: 9
        asize: 254980915200
        is_log: 0
        create_txg: 4
        children[0]:
            type: 'disk'
            id: 0
            guid: 11507650001572448411
            path: '/dev/disk/by-id/ata-Micron_1100_MTFDDAK256TBN_173718FA4342-part3'
            whole_disk: 0
            DTL: 643
            create_txg: 4
        children[1]:
            type: 'disk'
            id: 1
            guid: 8148307730435269824
            path: '/dev/disk/by-id/ata-Micron_1100_MTFDDAK256TBN_170515ACA5C0-part3'
            whole_disk: 0
            DTL: 133
            create_txg: 4
    features_for_read:
        com.delphix:hole_birth
        com.delphix:embedded_data
    labels = 0 1 2 3
```

Reply:
Ok. Also don't know about a better way just now.

Reply:
yeah, sadly lsblk has bad zfs support. thats also on of the reason why blkdeactivate what not enough